src/IIIFPresentation/API/Features/Manifest/DlcsManifestCoordinator.cs

```diff
@@ -83,30 +83,53 @@ public async Task<DlcsInteractionResult> HandleDlcsInteractions(
             
         }
 
+        return await UpdateDlcsWithAssets(request, manifestId, dbManifest, assets, spaceId, cancellationToken);
+    }
+
+    private async Task<DlcsInteractionResult> UpdateDlcsWithAssets(WriteManifestRequest request, string manifestId, 
```

Comment:
General observation about this class - it seems really complex. There's a lot happening - consider refactoring to simplify? As discussed on slack, would it be better to iterate over each CanvasPainting and attempt to categorise individually, rather than bulk categorisations? Starting with the most likely/least expensive to identify use case, shortcutting if we can?

Failing that adding code comments that just give a narrative description would help the reader